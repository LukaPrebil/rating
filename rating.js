export const rating = (n, options = { max: 5 }) =>
  ("★".repeat(options.max) + "☆".repeat(options.max)).slice(
    options.max - n,
    2 * options.max - n
  );
