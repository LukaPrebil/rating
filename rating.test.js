import { rating } from "./rating";

describe(`${rating.name} tests`, () => {
  test.each([0, 1, 2, 3, 4, 5])(
    "rating(%i) should return correct string",
    (n) => {
      expect(rating(n)).toMatchSnapshot();
    }
  );

  it("Should allow configured max value", () => {
    expect(rating(5, { max: 10 })).toMatchSnapshot();
  })

  it("Should allow configured success symbol", () => {
    expect(rating(2, { symbolSuccess: "x" })).toMatchSnapshot();
  })

  it("Should allow configured fail symbol", () => {
    expect(rating(3, { symbolFail: "o" })).toMatchSnapshot();
  })

  it("Should allow configured both symbols", () => {
    expect(rating(4, { symbolSuccess: "x", symbolFail: "o" })).toMatchSnapshot();
  })

  it("Should allow configured both symbols and max value", () => {
    expect(rating(5, { symbolSuccess: "x", symbolFail: "o", max: 10 })).toMatchSnapshot();
  })
});
