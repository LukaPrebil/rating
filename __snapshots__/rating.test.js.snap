// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rating tests Should allow configured both symbols 1`] = `"xxxxo"`;

exports[`rating tests Should allow configured both symbols and max value 1`] = `"xxxxxooooo"`;

exports[`rating tests Should allow configured fail symbol 1`] = `"★★★oo"`;

exports[`rating tests Should allow configured max value 1`] = `"★★★★★☆☆☆☆☆"`;

exports[`rating tests Should allow configured success symbol 1`] = `"xx☆☆☆"`;

exports[`rating tests rating(0) should return correct string 1`] = `"☆☆☆☆☆"`;

exports[`rating tests rating(1) should return correct string 1`] = `"★☆☆☆☆"`;

exports[`rating tests rating(2) should return correct string 1`] = `"★★☆☆☆"`;

exports[`rating tests rating(3) should return correct string 1`] = `"★★★☆☆"`;

exports[`rating tests rating(4) should return correct string 1`] = `"★★★★☆"`;

exports[`rating tests rating(5) should return correct string 1`] = `"★★★★★"`;
